autopush/db.py

```diff
@@ -432,42 +444,60 @@ def generate_last_connect_values(date):
             yield int(val)
 
 
-def list_tables(client=g_client):
-    """Return a list of the names of all DynamoDB tables."""
-    start_table = None
-    while True:
-        if start_table:  # pragma nocover
-            result = client.list_tables(ExclusiveStartTableName=start_table)
-        else:
-            result = client.list_tables()
-        for table in result.get('TableNames', []):
-            yield table
-        start_table = result.get('LastEvaluatedTableName', None)
-        if not start_table:
-            break
+def table_exists(tablename, boto_resource=None):
+    # type: (str, DynamoDBResource) -> bool
+    """Determine if the specified Table exists"""
+    try:
+        return DynamoDBTable(boto_resource, tablename).table_status in [
+            'CREATING', 'UPDATING', 'ACTIVE']
```

Comment:
Might want to use the Waiter thing to wait on creating/updating to finish, as other things may try to use the table immediately after this returns, which would fail if its not ACTIVE. That would only happen on startup I think, but could be problematic. Having a separate PR is fine.

Reply:
I'm not sure. `table_exists` is only called before a table creation command. In those situations. The command returns that the table either exists or is in the process of creation. 

`preflight_check()` already has a table ready wait routine, and is part of the startup for the process. 

Do you believe another delay would be useful?